src/k8s-engine/hyperkit.ts

```diff
@@ -231,6 +231,16 @@ export default class HyperkitBackend extends events.EventEmitter implements K8s.
     const { driver, defaultArgs } = this.hyperkitArgs;
 
     process.stderr.write(`\u001B[0;1m${ JSON.stringify([driver].concat(defaultArgs, args)) }\u001B[0m\n`);
+    if (args[0] === 'stop') {
+      // Fire and forget -- we're stopping and don't care about any errors.
+      // See issue 333 for full details.
+      // If we try to stop the driver in the following Promise code,
+      // it actually doesn't resolve until after the framework issues an
+      // UnhandledPromiseRejectionWarning warning.
+      childProcess.execFile(driver, defaultArgs.concat(args), options);
```

Comment:
Would it be better to emit a log instead of calling `reject` (about 10 lines down), so that we can still track when it fails?

Reply:
Added the log to the reject code, but I never see any output from that change.

And I agree with you on the change. It doesn't make total sense. But here's what I saw when I had more log statements:

* issue the stop command inside a promise
* something else emits the UnhandledPromiseException
* I then see the stop childprocess handle an exit with no 0-status.

  - I don't know what is emitting the warning, but it's like the nodejs shutdown apparatus has started firing before the asynch process exits and the promise gets resolved. By not running the code inside a promise, there's no unresolved process to deal with during shutdown.

* When I ran both processes with the `--unhandled-rejections=strict` flag I got the same result instead of a different, fatal error, or, even better, a traceback.

Reply:
This change is now obsolete.